### ISSUE TITLE
Update grid with similar images using vector search

### ISSUE THREAD

**Issue by akamil-etsy** [typesense/typesense-instantsearch-adapter#167]
## Description

is there a way to do something similar to https://same.energy/ by using typesense vector search, i.e. when user clicks on an image update the `hits` grid with similar images retrieved via ANN 

i.m aware of https://github.com/typesense/typesense-instantsearch-semantic-search-demo but it has different functionality with click on Find Similar opening a separate window

**Comment by jasonbosco** [comment]
I'm actually working on a demo that is going to have a very similar "Pinterest-style" UI. But I wasn't planning to use Instantsearch.js for it, given that it wasn't designed to do this natively... 

Any reason you want to use Instantsearch.js for it, vs building your own UI?

**Comment by akamil-etsy** [comment]
lazyness mostly. looking forward for your pinterest style demo!

**Comment by jasonbosco** [comment]
Ha! I just noticed a search bar on top of same.energy, which I wasn't planning on initially. I'll try to see if I can do this with instantsearch.js first. 

My dataset is going to be HN comments, so all text. But the concept is the same - take some data (in my case HN comments, in same.energy image), generate embeddings for it, index it in Typesense and then do an ANN search as each result (text / image) is clicked.

**Comment by akamil-etsy** [comment]
yeah, starting from text search and then refining visually is the most natural interaction it seems. The other option btw is starting from image and add text tags to refine

**Comment by jasonbosco** [comment]
I do like the idea of starting from text search and then drilling down from there visually.

**Comment by akamil-etsy** [comment]
btw same.energy has text demo for tweets as well https://t.same.energy/

**Comment by jasonbosco** [comment]
Oh interesting! Thank you for sharing
